native/src/snappyclient/cpp/impl/ControlConnection.cpp

```diff
@@ -69,10 +69,16 @@ ControlConnection::ControlConnection(ClientService * const &service) :
   m_controlHost = service->getCurrentHostAddress();
   boost::assign::insert(m_snappyServerTypeSet)(
       service->getServerType(true, false, false));
-  std::copy(m_locators.begin(), m_locators.end(),
-      std::inserter(m_controlHostSet, m_controlHostSet.end()));
+
+  addHosts(m_locators);
```

Comment:
You don't need a map here. The reason it is there in the java code is to get the "full" HostAddress including ServerType filled by server while the lookup up may have only partial fields set. So what is required is original host stored in the set which is not possible in Java Set API. In C++ find() already gives iterator which returns the original key in the set, so one needs to just get that.

Reply:
Reverted back to set from map